### PR TITLE
Create weather-updating module

### LINKED ISSUE #17: Create module which handles insert/update weather model
- Should be able to handle duplicated information
- Must ceate test for this module

### PR DESCRIPTION
**Pull request by yeareater**
Create module and test insert/update.
This resolves #17

### CODE REVIEW

File: tests/weather.updater.spec.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		otherWeather = rhs.weathers[index];
+		should.equal(weather.hour, otherWeather.hour);
+		should.equal(weather.temp, otherWeather.temp);
+		should.equal(weather.skyCode, otherWeather.skyCode);
+		should.equal(weather.rainfallCode, otherWeather.rainfallCode);
+		should.equal(weather.rainfallProbability, otherWeather.rainfallProbability);
+	});
+}
+
+function checkWeathersHaveProperties (weathersArray) {
```
Comment (DonghyunNa) [inline tests/weather.updater.spec.js]: need unified indent

**Review by DonghyunNa — CHANGES_REQUESTED** [review]
need same type of indent in your code

File: models/weatherUpdater.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+const Weather = require("./weatherSchema");
+const weatherApi = require("./weather-api");
+
+module.exports.updateWeather = function(weathers) {
```
Comment (crowjdh) [inline models/weatherUpdater.js]: How about changing name of the function into **saveWeathers** and the file name into **weather-utils**?

File: models/weatherUpdater.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+				}
+				if (weathers.length == ++savedCount) {
+					resolve();
+				}
+			});
+		}
+	});
+}
+
+function updateWeather(dailyWeather, callback) {
```
Comment (crowjdh) [inline models/weatherUpdater.js]: This function performs *save&update* at the same time. How about changing name of the function into **saveOrUpdateWeather**?

File: models/weatherUpdater.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+		}
+		if(!oldWeather){
+			saveNewDailyWeathers(dailyWeather, callback);
+		} else {
+			updateHourlyWeather(oldWeather, dailyWeather, callback);
+		}
+	});
+}
+
+function saveNewDailyWeathers(newDailyWeather, callback) {
```
Comment (crowjdh) [inline models/weatherUpdater.js]: Suggestions:
- function saveWeather(weather, callback) { ... }

File: models/weatherUpdater.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+			console.log("Save err");
+			callback(err);
+		} else {
+			console.log("Weather saved");
+			callback();
+		}
+	});
+}
+
+function updateHourlyWeather(oldWeather, newWeather, callback) {
```
Comment (crowjdh) [inline models/weatherUpdater.js]: I think the name of the function can be **updateWeather**.

File: tests/weather.updater.spec.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+			hour: (3*hourBase) + (3*i),
+			temp: hourBase + i + day,
+			skyCode: hourBase + i + day,
+			rainfallCode: hourBase + i + day,
+			rainfallProbability: hourBase + i + day
+		});
+	}
+
+	return weather;
+
```
Comment (crowjdh) [inline tests/weather.updater.spec.js]: Redundant new line here :)

File: tests/weather.updater.spec.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+			should.not.exist(err);
+			weatherUpdater.updateWeather([secondWeather]).then(function(err) {
+				should.not.exist(err);
+				Weather.findOne({ date: firstWeather.date }, function(err, updatedWeather) {
+					should.not.exist(err);
+					compare(updatedWeather, weather);
+					done();
+				});
+			});
+		});
```
Comment (crowjdh) [inline tests/weather.updater.spec.js]: Lines at 44~54 should be:
```javascript
weatherUtil.saveWeathers([firstWeather]).then(function(err) {
    should.not.exist(err);
    return weatherUtil.saveWeathers([secondWeather]);
}).then(function(err) {
    should.not.exist(err);
    Weather.findOne({ date: firstWeather.date }, function(err, updatedWeather) {
        should.not.exist(err);
        assertEquals(updatedWeather, weather);
        done();
    });
});
```

File: tests/weather.updater.spec.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+	});
+
+});
+
+function compare(lhs, rhs) {
```
Comment (crowjdh) [inline tests/weather.updater.spec.js]: Changing name of this function into **assertEquals** seems better :)

**Review by crowjdh — CHANGES_REQUESTED** [review]
Great job :)
I left some minor(?) improvements you can apply.

Also, consider removing "console.log"s where they're unnecessary.

**Comment by crowjdh** [comment]
Also, feel free to comment to my reviews if it doesn't seem legitimate.(Doubt everything!!)

**Review by crowjdh — APPROVED** [review]

**Review by DonghyunNa — APPROVED** [review]
good

**Pull request merged**